py/jpy-integration/build.gradle

```diff
@@ -82,54 +83,129 @@ idea {
+
+
+  Sync jpyConfigTask = tasks.create 'createPythonToJavaConfig', Sync, {
+    // if the classpath changes, we need to rebuild this
+    inputs.files sourceSets.pythonToJava.runtimeClasspath
+    from('src/pythonToJava/jpyconfig.py.template')
+    into "${buildDir}/jpyConfig"
+    rename { file -> 'jpyconfig.py' }
+    doFirst {
+      expand(
+              javaHome:'/opt/java/openjdk',
+              jvmMaxMem:'32m',
+              jvmClasspath:'/classpath:' + sourceSets.pythonToJava.runtimeClasspath.asList().collect { '/classpath/' + it.name }.join(':')
+      )
+    }
   }
 
-  tasks.create 'testPython', {
-    Task t ->
-      t.group = 'python'
-      t.description = 'Run all python-related tests in the jpy-integration module'
-      check.dependsOn t
-      t.dependsOn(pythonTests)
+  Closure<TaskProvider<Task>> pyExec = { String taskName, List<String> command, File reportDir = project.file("${buildDir}/test-results/${taskName}")->
+    return Docker.registerDockerTask(project, taskName) {
+      copyIn {
+        from ('src/pythonToJava/python') {
+          into 'python'
+        }
+        from (sourceSets.pythonToJava.runtimeClasspath) {
+          into 'classpath'
+        }
+        from(jpyConfigTask.outputs.files) {
+          into 'python'
+        }
+      }
+      parentContainers = [project(':Integrations').tasks.findByName('buildDeephavenPython')] // deephaven/runtime-base
+      imageName = 'deephaven/jpy-integration-python-to-java-tests'
+      dockerfile {
+        // set up the container, env vars - things that aren't likely to change
+        from 'deephaven/runtime-base'
+        runCommand '''set -eux; \\
+                        pip3 install unittest-xml-reporting==3.0.4;\\
+                        mkdir /out;'''
+        environmentVariable 'JDK_HOME', '/opt/java/openjdk'
+        environmentVariable 'JAVA_VERSION', '1.8'
+        environmentVariable 'JPY_PY_CONFIG', '/python/jpyconfig.py'
+
+        workingDir '/python'
+
+        // copy in the contents that we do expect to change as the project updates
+        copyFile 'python', '/python'
+        copyFile 'classpath', '/classpath'
+      }
+      entrypoint = command
+
+      copyOut {
+        into reportDir
+      }
+    }
   }
 
+  pythonToJava.dependsOn pyExec('unittest-python-to-java', ['python3', '-m', 'xmlrunner', 'discover', '-v', '-o', '/out/'])
+
+  // These tests need to run in their own process, since jpy can only start up a jvm once per process.
+  pythonToJava.dependsOn pyExec('unittest-single-jpy-python-to-java', ['python3', '-m', 'xmlrunner', '-v', '-o', '/out/', 'test.main_single_jpy_create_destroy_jvm'])
+  pythonToJava.dependsOn pyExec('unittest-multiple-jpy-python-to-java', ['python3', '-m', 'xmlrunner', '-v', '-o', '/out/', 'test.main_multiple_jpy_create_destroy_jvm'])
+
+  pythonToJava.dependsOn pyExec('debugPythonToJava', ['python3', 'java_debug.py'])
```

Comment:
This task would benefit from `showLogsOnSuccess = true`.

Reply:
agreed, will add - that flag was added very late in this patch, didnt consider where else to put it yet.